hoomd/md/pytest/test_pair.py

```diff
@@ -0,0 +1,939 @@
+    for key in param_dict1:
+        assert param_dict1[key] == param_dict2[key]
+
+
+def test_rcut(simulation_factory, two_particle_snapshot_factory):
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), r_cut=2.5)
+    lj.params[('A', 'A')] = {'sigma': 1, 'epsilon': 0.5}
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_cut[('A', 'A')] = 'str'
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_cut[('A', 'A')] = [1, 2, 3]
+
+    sim = simulation_factory(two_particle_snapshot_factory(dimensions=3, d=.5))
+    integrator = hoomd.md.Integrator(dt=0.005)
+    integrator.forces.append(lj)
+    integrator.methods.append(hoomd.md.methods.Langevin(hoomd.filter.All(),
+                                                        kT=1, seed=1))
+    sim.operations.integrator = integrator
+
+    _assert_equivalent_type_params(lj.r_cut.to_dict(), {('A', 'A'): 2.5})
+    sim.operations.schedule()
+    sim.run(1)
+    _assert_equivalent_type_params(lj.r_cut.to_dict(), {('A', 'A'): 2.5})
+
+
+@pytest.mark.parametrize("mode", ['none', 'shifted', 'xplor'])
+def test_mode(simulation_factory, two_particle_snapshot_factory, mode):
+    cell = hoomd.md.nlist.Cell()
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode=1)
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode='str')
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode=[1, 2, 3])
+
+    lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode=mode)
+    lj.params[('A', 'A')] = {'sigma': 1, 'epsilon': 0.5}
+    snap = two_particle_snapshot_factory(dimensions=3, d=.5)
+    sim = simulation_factory(snap)
+    integrator = hoomd.md.Integrator(dt=0.005)
+    integrator.forces.append(lj)
+    integrator.methods.append(hoomd.md.methods.Langevin(hoomd.filter.All(),
+                                                        kT=1, seed=1))
+    sim.operations.integrator = integrator
+    sim.operations.schedule()
+    sim.run(1)
+
+
+def test_ron(simulation_factory, two_particle_snapshot_factory):
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), mode='xplor', r_cut=2.5)
+    lj.params[('A', 'A')] = {'sigma': 1, 'epsilon': 0.5}
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_on[('A', 'A')] = 'str'
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_on[('A', 'A')] = [1, 2, 3]
+
+    sim = simulation_factory(two_particle_snapshot_factory(dimensions=3, d=.5))
+    integrator = hoomd.md.Integrator(dt=0.005)
+    integrator.forces.append(lj)
+    integrator.methods.append(hoomd.md.methods.Langevin(hoomd.filter.All(),
+                                                        kT=1, seed=1))
+    sim.operations.integrator = integrator
+    assert lj.r_on.to_dict() == {}
```

Comment:
I would set `ron` after this to check that attaching works correctly on it. We can then set it to a new value if we want to test setting after attaching.